src/frontends/pytorch/src/transforms/irfftn_complex_replacer.cpp

```diff
@@ -0,0 +1,159 @@
+        std::shared_ptr<ov::Node> dim;
+        if (!dim_use_default) {
+            // Dim values is provided, load from input.
+            dim = std::make_shared<v0::Convert>(concat_list_construct(irfftn_op->input_value(2)), element::i32);
+        } else if (!s_use_default) {
+            // If dim is default and s is provided, use last s_len dimensions where s_len is length of s.
+            auto s_len = std::make_shared<v3::ShapeOf>(raw_s_input_maybe, element::i32);
+            auto range_start = std::make_shared<v1::Subtract>(input_rank, s_len);
+            auto range_start_scalar = std::make_shared<v0::Squeeze>(range_start);
+            dim = std::make_shared<v4::Range>(range_start_scalar, input_rank_scalar, const_scalar_1, element::i32);
+        } else {
+            // Dim and s are set to default, use all of dimensions.
+            dim = std::make_shared<v4::Range>(const_scalar_0, input_rank_scalar, const_scalar_1, element::i32);
+        }
+
+        // Calculate default s values. Use full available size except last element, which is set to even value in last
+        // dimension: s[-1] = 2 * (complex_input_shape[dim[-1]])
+        auto default_s_raw = std::make_shared<v8::Gather>(complex_input_shape, dim, const_0);
+        auto last_s = std::make_shared<v8::Gather>(default_s_raw, const_neg_1, const_0);
+        auto last_s_m_1 = std::make_shared<v1::Subtract>(last_s, const_1);
+        auto s_upd = std::make_shared<v1::Multiply>(last_s_m_1, const_2);
+        auto s_shape = std::make_shared<v3::ShapeOf>(default_s_raw, element::i32);
+        auto last_s_idx = std::make_shared<v1::Subtract>(s_shape, const_1);
+        auto default_s = std::make_shared<v3::ScatterUpdate>(default_s_raw, last_s_idx, s_upd, const_0);
+
+        // Handle s parameter containing vector of intigers indicating signal sizes for dimensions.
+        std::shared_ptr<ov::Node> s;
+        if (!s_use_default) {
+            // Values for s were provided. Replace -1 values with default full size in given dimension.
+            auto full_s_cond = std::make_shared<v1::Equal>(raw_s_input_maybe, const_neg_1);
+            s = std::make_shared<v1::Select>(full_s_cond, default_s, raw_s_input_maybe);
+        } else {
+            // Value for s was set to default.
+            s = default_s;
+        }
+
+        // Handle norm parameter indicating normalization mode to use. Defaults to "backward".
+        std::string norm;
+        if (const auto& fw_node_mode = std::dynamic_pointer_cast<ov::op::util::FrameworkNode>(
+                irfftn_op->input_value(3).get_node_shared_ptr())) {
+            const auto& attrs = fw_node_mode->get_attrs();
+            if (attrs.find("string_value") != attrs.end()) {
+                norm = attrs.at("string_value");
+            } else {
+                norm = "backward";
+            }
+        } else {
+            return false;
```

Comment:
Due to merge of #18015 please add some error message using `add_exception_to_fw_node` in cases when your transformation should've worked, but didn't.

Reply:
Done